cmd/lava/internal/run/run.go

```diff
@@ -51,17 +51,16 @@ func run(args []string) error {
 
 	base.LogLevel.Set(cfg.LogLevel)
 
-	report, err := engine.Run(cfg.ChecktypesURLs, cfg.Targets, cfg.AgentConfig)
+	er, err := engine.Run(cfg.ChecktypesURLs, cfg.Targets, cfg.AgentConfig)
 	if err != nil {
 		return fmt.Errorf("run: %w", err)
 	}
 
-	// TODO(rm): show report and exit with the proper exit code.
-	enc := json.NewEncoder(os.Stdout)
-	enc.SetIndent("", "  ")
-	if err := enc.Encode(report); err != nil {
-		return fmt.Errorf("encode report: %w", err)
+	r := report.NewWriter(cfg.ReportConfig)
+	exitCode, err := r.Write(er)
```

Comment:
I think this is fine:

```suggestion
	exitCode, err := report.NewWriter(cfg.ReportConfig).Write(er)
```

Reply:
I can't do this in one single line because now `NewWriter` is returning the exitCode and an error.

Reply:
Ah, right. And now we also have to call `defer r.Close()` if the returned error is not nil.

Reply:
BTW now that the type is `report.Writer`, I would probably rename `r` to `rw`.